Add ability to run cluster of servers

Will-Howard opened the pull request.

jpaddison3 commented:
> We should probably silence this error.

We have [just silenced this error](https://github.com/ForumMagnum/ForumMagnum/pull/6605).

jpaddison3 commented:
@jimrandomh and I were just talking about this.

oetherington commented on packages/lesswrong/server/sqlConnection.ts:
```diff
@@ -113,7 +120,13 @@ export const createSqlConnection = async (url?: string): Promise<SqlClient> => {
+    await Promise.all(onConnectQueries.map((query) => {
+      return db.tx(async (transaction) => {
+        // Set advisory lock to ensure only one server runs each query at a time
+        await transaction.any(`SELECT pg_advisory_xact_lock(${getLockKey(query)});`);
```
Just to be safe, we should probably set a timeout with `SET LOCAL lock_timeout = '5s'` (or however long we're willing to wait).

oetherington reviewed (approved):
Looks good to me! I guess we should probably discuss this in a dev check in at some point before pulling the trigger, but I think I'm probably in favour of it.

oetherington commented:
A couple of other open questions:
 - stdout is kind of a mess - I think this would be dependant on having some better log handling
 - We initalize the database clients before forking and they seem to still work after being copied into the forks, but this seems risky to me and I don't have a good mental model of whether or not I should _expect_ this to work.
 - I see a ton of `Cannot read properties of undefined (reading 'getIndex')` 15 seconds after the server starts when we try to create all the indexes - I'm guessing this is also something related to initializing globals before forking.

jpaddison3 commented:
I'm going to remove the due date from the asana task here as it's an agreed upon low urgency PR

Will-Howard commented:
Ok I've updated this and addressed all (I think of the issues):
 - I've made it fork earlier in the startup sequence so each process gets it's own sql client
   -  This also fixed the index creation problem Ollie mentioned above
 - I've added the process id to the logs

I'm going to merge this and turn it on on staging for testing, then I'll turn it on for the bot environment, and then actual prod if there are no issues
